libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizerOptionsV2.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime;
+using Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime.Models;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.AI.Luis
+{
+    public class LuisRecognizerOptionsV2 : LuisRecognizerOptions
+    {
+        /// <summary>
+        /// The context label for a LUIS trace activity.
+        /// </summary>
+        public const string LuisTraceLabel = "Luis Trace";
+
+        /// <summary>
+        /// The value type for a LUIS trace activity.
+        /// </summary>
+        public const string LuisTraceType = "https://www.luis.ai/schemas/trace";
+
+        public LuisRecognizerOptionsV2(LuisApplication application)
+        : base(application) 
+        {
+        }
+
+        public LuisPredictionOptions PredictionOptions { get; set; } = new LuisPredictionOptions();
+
+        internal override async Task<RecognizerResult> RecognizeInternalAsync(ITurnContext turnContext, HttpClient httpClient,  CancellationToken cancellationToken)
+        {
+            BotAssert.ContextNotNull(turnContext);
+
```

Comment:
Shoud there be a check for Activity not being null as well? Given that turnContext implement an interface I would assume the implementation could be that of a third party and we should not assume it is not null without verifying.

(and there's code later on that has turnContext.Activity?. ..., so I assume that it can be null, so we should check here or make the next line not crash if it is null)